ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -173,70 +220,169 @@ func (o *AviObjectGraph) BuildPGPool(key, parentNsName string, childVsNode *node
+		//TODO(Akshay): add condition here.
+		return pgAttachedToVS
+	}
+
+	allListeners := akogatewayapiobjects.GatewayApiLister().GetRouteToGatewayListener(routeTypeNsName)
+	listeners := []akogatewayapiobjects.GatewayListenerStore{}
+	for _, listener := range allListeners {
+		if listener.Gateway == parentNsName {
+			listeners = append(listeners, listener)
+		}
+	}
+	//ListenerName/port/protocol/allowedRouteSpec
+	listenerProtocol := listeners[0].Protocol
+
+	PG := &nodes.AviPoolGroupNode{
+		Name:   PGName,
+		Tenant: lib.GetTenant(),
+	}
+	for _, backend := range rule.Backends {
+		poolName := akogatewayapilib.GetPoolName(parentNs, parentName,
+			routeModel.GetNamespace(), routeModel.GetName(),
+			matchName,
+			backend.Backend.Namespace, backend.Backend.Name, strconv.Itoa(int(backend.Backend.Port)))
+		svcObj, err := utils.GetInformers().ServiceInformer.Lister().Services(backend.Backend.Namespace).Get(backend.Backend.Name)
+		if err != nil {
+			utils.AviLog.Debugf("key: %s, msg: there was an error in retrieving the service", key)
+			o.RemovePoolRefsFromPG(poolName, o.GetPoolGroupByName(PGName))
+			continue
+		}
+		poolNode := &nodes.AviPoolNode{
+			Name:       poolName,
+			Tenant:     lib.GetTenant(),
+			Protocol:   listenerProtocol,
+			PortName:   akogatewayapilib.FindPortName(backend.Backend.Name, backend.Backend.Namespace, backend.Backend.Port, key),
+			TargetPort: akogatewayapilib.FindTargetPort(backend.Backend.Name, backend.Backend.Namespace, backend.Backend.Port, key),
+			Port:       backend.Backend.Port,
+			ServiceMetadata: lib.ServiceMetadataObj{
+				NamespaceServiceName: []string{backend.Backend.Namespace + "/" + backend.Backend.Name},
+			},
+			VrfContext: lib.GetVrf(),
+		}
+		poolNode.NetworkPlacementSettings = lib.GetNodeNetworkMap()
+		serviceType := lib.GetServiceType()
+		if serviceType == lib.NodePort {
+			servers := nodes.PopulateServersForNodePort(poolNode, svcObj.ObjectMeta.Namespace, svcObj.ObjectMeta.Name, false, key)
+			if servers != nil {
+				poolNode.Servers = servers
 			}
-			if !dataScriptRefExists {
-				childVsNode.HTTPDSrefs = append(childVsNode.HTTPDSrefs, dsScriptNode)
+		} else {
+			servers := nodes.PopulateServers(poolNode, svcObj.ObjectMeta.Namespace, svcObj.ObjectMeta.Name, false, key)
+			if servers != nil {
+				poolNode.Servers = servers
 			}
 		}
+		if parentVsNode.CheckPoolNChecksum(poolNode.Name, poolNode.GetCheckSum()) {
+			// Replace the poolNode.
+			parentVsNode.ReplaceEvhPoolInEVHNode(poolNode, key)
+			// Add pool to list of Pools attached to Parent VS
+			httppsPGPool.Pool = append(httppsPGPool.Pool, poolNode.Name)
+			uniquePools := sets.NewString(httppsPGPool.Pool...)
+			httppsPGPool.Pool = uniquePools.List()
+		}
 		pool_ref := fmt.Sprintf("/api/pool?name=%s", poolNode.Name)
-		ratio := uint32(httpbackend.Backend.Weight)
+		o.BuildBackendFiltersModel(key, poolName, backend, parentVsNode)
```

Comment:
Please check if backendRefFilters flow is working as expected in this case as well since in the former case HTTPRoute with hostname we attach the datascript to the childVs, however here we add it to the parentVs

Reply:
Is this validated ?

Reply:
bakendref changes are on hold so not tested and not enabled it also.

Reply:
okay